database/transaction.rs

```diff
@@ -314,25 +335,20 @@ impl<D: DurabilityClient> TransactionSchema<D> {
     }
 }
 
-#[derive(Debug, Clone)]
-pub enum SchemaCommitError {
-    ConceptWrite { errors: Vec<ConceptWriteError> },
-    TypeCacheUpdate { source: TypeCacheCreateError },
-    Statistics { source: StatisticsError },
-    FunctionError { source: FunctionError },
-    SnapshotError { source: SnapshotError },
-}
-
-impl fmt::Display for SchemaCommitError {
-    fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        todo!()
+// TODO: Same issue with ConceptWriteErrors vs ErrorsFirst as for DataCommitError
+typedb_error!(
+    pub SchemaCommitError(component = "Schema commit", prefix = "SCT") {
+        ConceptWriteErrors(1, "Schema commit error.", source: Vec<ConceptWriteError> ),
+        ConceptWriteErrorsFirst(2, "Schema commit error.", ( typedb_source : ConceptWriteError )),
+        TypeCacheUpdateError(3, "TypeCache update error.", ( typedb_source: TypeCacheCreateError )),
+        StatisticsError(4, "Statistics error.", ( typedb_source: StatisticsError )),
+        FunctionError(5, "Function error.", ( typedb_source: FunctionError )),
+        SnapshotError(6, "Snapshot error.", ( typedb_source: SnapshotError )),
     }
-}
+);
 
-impl Error for SchemaCommitError {
-    fn source(&self) -> Option<&(dyn Error + 'static)> {
-        todo!()
+typedb_error!(
+    pub TransactionError(component = "Transaction", prefix = "TXN") {
+        Timeout(1, "Transaction timeout.", source: RecvTimeoutError),
     }
-}
-
-// TODO: TypeDB Error
+);
```

Comment:
Item-like macros should use `{}` rather than `()`.
`{}`, `[]`, and `()` are interchangeable in macro invocations, but the convention is `{}` for item-like (e.g. `thread_local! {}`), `()` for function-like (`print!()`), `[]` for array-like (`vec![]`).

Reply:
@flyingsilverfin thoughts?

Reply:
@flyingsilverfin ping